jerry-core/jrt/jrt.h

```diff
@@ -108,7 +113,11 @@ extern void __noreturn jerry_unreachable (const char *, const char *, const uint
     } \
   } while (0)
 
+#ifdef(__GNUC__)
 #define JERRY_UNREACHABLE() __builtin_unreachable ()
+#else /* !__GNUC__ */
+#define JERRY_UNREACHABLE() abort(); /* but what if they haven't included stdlib?!? */
```

Comment:
One thing I forgot to mention: The key difference between `__builtin_unreachable()` and `abort()` is that calling `__builtin_unreachable()` yields undefined behavior while calling `abort()` is perfectly valid, e.g. not triggering any kind of undefined behavior. The compiler obviously can optimize much more aggressively when undefined behavior is involved. However, since the goal here is to have a solution that works in a standard C99 environment without relying on any compiler-specific built-in functions, `abort()` seems to be the closest match to `__builtin_unreachable()` in terms of intended behavior. Losing some optimization opportunities is essentially the price that needs to be paid to get a solution that works in any standard C99 environment.